jfrog-cli/artifactory/utils/golang/project/dependencies/dependenciesutils.go

```diff
@@ -0,0 +1,452 @@
+	moduleId = strings.Join(moduleInfo, ":")
+	modulePath := strings.Replace(moduleId, ":", "@", 1)
+	path = filepath.Join(tempDir, modulePath, "go.mod")
+
+	multiReader, err := multifilereader.NewMultiFileReaderAt([]string{dep.GetZipPath()})
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	err = fileutils.Unzip(multiReader, multiReader.Size(), tempDir)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	err = ioutil.WriteFile(path, dep.GetModContent(), 0700)
+	if err != nil {
+		return path, errorutils.CheckError(err)
+	}
+	return path, nil
+}
+
+func replaceExclamationMarkWithUpperCase(moduleName string) string {
+	var str string
+	for i := 0; i < len(moduleName); i++ {
+		if string(moduleName[i]) == "!" {
+			if i < len(moduleName)-1 {
+				r := rune(moduleName[i+1])
+				str += string(unicode.ToUpper(r))
+				i++
+			}
+		} else {
+			str += string(moduleName[i])
+		}
+	}
+	return str
+}
+
+// Runs the go mod download command. Should set first the environment variable of GoProxy
+func downloadDependency(downloadFromArtifactory bool, fullDependencyName, targetRepo string, details *config.ArtifactoryDetails) error {
+	var err error
+	if downloadFromArtifactory {
+		log.Debug("Downloading dependency from Artifactory:", fullDependencyName)
+		err = golang.SetGoProxyEnvVar(details, targetRepo)
+	} else {
+		log.Debug("Downloading dependency from VCS:", fullDependencyName)
+		err = os.Unsetenv(golang.GOPROXY)
+	}
+	if errorutils.CheckError(err) != nil {
+		return err
+	}
+
+	err = golang.DownloadDependency(fullDependencyName)
+	return errorutils.CheckError(err)
+}
+
+func populateModAndGetDependenciesGraph(path string, shouldRunGoModCommand, shouldRunGoGraph bool) (output map[string]bool, err error) {
+	err = os.Chdir(filepath.Dir(path))
+	if errorutils.CheckError(err) != nil {
+		return
+	}
+	log.Debug("Preparing to populate mod", filepath.Dir(path))
+	// Remove go.sum file to avoid checksum conflicts with the old go.sum
+	goSum := filepath.Join(filepath.Dir(path), "go.sum")
+	exists, err := fileutils.IsFileExists(goSum, false)
+	if err != nil {
+		return
+	}
+
+	if exists {
+		err = os.Remove(goSum)
+		if errorutils.CheckError(err) != nil {
+			return
+		}
+	}
+
+	if shouldRunGoModCommand {
+		// Running go mod tidy command
+		err = golang.RunGoModTidy()
+		if err != nil {
+			return
+		}
+	}
+	if shouldRunGoGraph {
+		// Running go mod graph command
+		output, err = golang.GetDependenciesGraph()
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+// Downloads the mod file from Artifactory to the Go cache
+func downloadModFileFromArtifactoryToLocalCache(cachePath, targetRepo, name, version string, details *config.ArtifactoryDetails, client *httpclient.HttpClient) string {
+	pathToModuleCache := filepath.Join(cachePath, name, "@v")
+	dirExists, err := fileutils.IsDirExists(pathToModuleCache, false)
+	if err != nil {
+		log.Debug("Received and error:", err)
```

Comment:
Typo and suggest suggest changing to log.Error.
I'll be skipping the rest from now...so feel free to search for the other places :)

Reply:
Done